src/Conversion/ONNXToTOSA/Tensor/PaddingOp.cpp

```diff
@@ -0,0 +1,169 @@
+//         tensor of bfloat16 type values or tensor of 16-bit/32-bit/64-bit
+//         float values or tensor of string type values or tensor of 1-bit
+//         signless integer values or tensor of complex type with
+//         32-bit/64-bit float elements values or memref of any type values
+//         or none type.
+//
+// Results:
+//  output
+//         tensor of 8-bit/16-bit/32-bit/64-bit unsigned integer values or
+//         tensor of 8-bit/16-bit/32-bit/64-bit signless integer values or
+//         tensor of bfloat16 type values or tensor of 16-bit/32-bit/64-bit
+//         float values or tensor of string type values or tensor of 1-bit
+//         signless integer values or tensor of complex type with
+//         32-bit/64-bit float elements values or memref of any type values
+//         or none type.
+//
+
+namespace onnx_mlir {
+
+class ONNXPadOpLoweringToTOSA : public OpConversionPattern<ONNXPadOp> {
+public:
+  using OpConversionPattern<ONNXPadOp>::OpConversionPattern;
+  using OpAdaptor = typename ONNXPadOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXPadOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op.getLoc();
+
+    Value data = adaptor.data();
+    Value pads = adaptor.pads();
+    Value constValue = adaptor.constant_value();
+
+    if (!(adaptor.mode() == "constant")) {
+      return rewriter.notifyMatchFailure(
+          op, "Only 'constant' mode is supported");
+    }
+
+    // creating the DenseElementsAttr using pads values.
+    DenseElementsAttr denseAttr =
+        llvm::dyn_cast_or_null<tosa::ConstOp>(pads.getDefiningOp())
+            .getValue()
+            .dyn_cast<DenseElementsAttr>();
+
+    // Reading the ONNX side pads values and store in the array.
+    llvm::SmallVector<APInt, 8> intValues;
+    bool paddingNeeded = false;
+    for (auto n : denseAttr.getValues<APInt>()) {
+      intValues.push_back(n);
+      if (!n.isZero())
+        paddingNeeded = true;
+    }
+    if (!paddingNeeded) {
+      // We do not need to represent the no-op pad in the resulting MLIR
+      rewriter.eraseOp(pads.getDefiningOp());
+      rewriter.eraseOp(constValue.getDefiningOp());
+      rewriter.replaceOp(op, {data});
+      return success();
+    }
+
+    // Rearrange the pad values.
+    // ONNX : [b1, b2, b3, b4, e1, e2, e3, e4]
+    // TOSA :[[b1, e1], [b2, e2], [b3, e3], [b4, e4]]
+    llvm::SmallVector<int64_t, 8> translatePadsList;
+    if (!intValues.empty()) {
+      const unsigned int dimSize = intValues.size() / 2;
+      for (unsigned int i = 0; i < dimSize; i++) {
+        translatePadsList.push_back(intValues[i].getZExtValue());
+        translatePadsList.push_back(intValues[i + dimSize].getZExtValue());
+      }
+    }
+
+    const unsigned int numberOfDims = intValues.size() / 2;
+    DenseElementsAttr paddingAttr = DenseIntElementsAttr::get(
+        RankedTensorType::get({numberOfDims, 2}, rewriter.getI64Type()),
+        translatePadsList);
```

Comment:
[TosaToLinalg](https://github.com/llvm/llvm-project/blob/3f81126a338796427a364719f19076d1a37e4b25/mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp#L231-L237) helped me. It was more a problem of not to mix up double and float in translatePadsList, because this was causing runtime errors without proper explanation